File: app/src/main/java/de/bornholdtlee/defaultproject/utils/HockeyAppUtils.java
package de.bornholdtlee.defaultproject.utils;

import android.app.Activity;

import net.hockeyapp.android.CrashManager;
import net.hockeyapp.android.CrashManagerListener;

import de.bornholdtlee.defaultproject.BuildConfig;
import lombok.Getter;

public final class HockeyAppUtils {

    private HockeyAppUtils() {
    }

    /**
     * Info:
     * Diese Abfrage sollte nicht in der Splashscreen-Activity ausgeführt werden,
     * sondern erst in der darauf folgenden Activity
     */
    public static void checkForCrashes(Activity activity) {
        CrashManager.register(activity, BuildConfig.HOCKEY_APP_ID);
    }

    public static void checkForCrashes(Activity activity, CrashListener listener) {
        CrashManager.register(activity, BuildConfig.HOCKEY_APP_ID, listener);
    }

    public abstract static class CrashListener extends CrashManagerListener {

        @Getter
        private boolean crashFound;

        @Override
        public void onNewCrashesFound() {
            crashFound = true;
        }

        @Override
        public void onConfirmedCrashesFound() {
            crashFound = false;
            onDialogHandled();
        }

        @Override
        public void onCrashesSent() {
            crashFound = false;
            onDialogHandled();
        }

        @Override
        public void onCrashesNotSent() {
            crashFound = false;
            onDialogHandled();
        }

        @Override
        public void onUserDeniedCrashes() {
            crashFound = false;
            onDialogHandled();
        }

        public abstract void onDialogHandled();
    }
}

File: app/src/main/java/de/bornholdtlee/defaultproject/utils/HockeyAppUtils.kt
package de.bornholdtlee.defaultproject.utils

import android.app.Activity

import net.hockeyapp.android.CrashManager
import net.hockeyapp.android.CrashManagerListener

import de.bornholdtlee.defaultproject.BuildConfig

object HockeyAppUtils {

    /**
     * Info:
     * Diese Abfrage sollte nicht in der Splashscreen-Activity ausgeführt werden,
     * sondern erst in der darauf folgenden Activity
     */
    fun checkForCrashes(activity: Activity) {
        CrashManager.register(activity, BuildConfig.HOCKEY_APP_ID)
    }

    fun checkForCrashes(activity: Activity, listener: CrashListener) {
        CrashManager.register(activity, BuildConfig.HOCKEY_APP_ID, listener)
    }

    abstract class CrashListener : CrashManagerListener() {

        private var crashFound: Boolean = false

        override fun onNewCrashesFound() {
            crashFound = true
        }

        override fun onConfirmedCrashesFound() {
            crashFound = false
            onDialogHandled()
        }

        override fun onCrashesSent() {
            crashFound = false
            onDialogHandled()
        }

        override fun onCrashesNotSent() {
            crashFound = false
            onDialogHandled()
        }

        override fun onUserDeniedCrashes() {
            crashFound = false
            onDialogHandled()
        }

        abstract fun onDialogHandled()
    }
}